src/components/app.js

```diff
@@ -180,17 +182,17 @@ const App = () => {
                     { /* to force drawer content to start below appbar */ }
                     <div className={classes.toolbar} />
                     <NetworkExplorer network={ network }
-                                     onSubstationClick={ id => showVoltageLevelDiagram(id) }/>
+                                     onSubstationClick={ (id, name) => showVoltageLevelDiagram(id, name) }/>
                 </Drawer>
                 <NetworkMap network={ network }
                             labelsZoomThreshold={8}
                             initialPosition={[2.5, 46.6]}
                             initialZoom={6}
-                            onSubstationClick={ id => showVoltageLevelDiagram(id) } />
+                            onSubstationClick={ (id, name) => showVoltageLevelDiagram(id, name) } />
                 {
                     diagram &&
                     <div style={{ position: "absolute", left: 250, top: 10, zIndex: 1 }}>
-                        <SingleLineDiagram diagramId={diagram.id} svg={ diagram.svg } />
+                        <SingleLineDiagram diagramName={diagram.name} diagramId={diagram.id} svg={ diagram.svg } />
```

Comment:
Passing id and name is useless as for the svg data,when mode name/id is changed the component is re-rendered.
You should only keep one property "diagramTitle" fed by id or name.